packages/versioning/lib/versioning.cadl

```diff
@@ -0,0 +1,213 @@
+import "../dist/src/versioning.js";
+
+#suppress "projections-are-experimental"
+projection op#v {
+  to(version) {
+    if addedAfter(self, version) {
+      return never;
+    } else if removedOnOrBefore(self, version) {
+      return never;
+    } else {
+      self.projectParameters(v, version);
```

Comment:
If by "handling decorators" you mean versioning how applied decorators and their arguments change over time, the framework itself doesn't handle that yet. @johanste had some proposal that was like `appliedOn(version, decorator, decoratorArgs)`, though this is maybe difficult to use/reason about/tool. Otherwise you could do it with a manual projection, which might be ok if this sort of thing is rare.